scipy/spatial/transform/_rigid_transformation.pyx

```diff
@@ -0,0 +1,1953 @@
+
+        An index array returns a stack of transformations:
+
+        >>> tf[[0, 2]].translation
+        array([[0., 0., 0.],
+               [2., 0., 0.]])
+        """
+        if self._single:
+            raise TypeError("Single transformation is not subscriptable.")
+
+        # Convert memoryview to numpy array before indexing:
+        arr = np.asarray(self._matrix)
+        return self.__class__(arr[indexer], copy=False)
+
+    @cython.embedsignature(True)
+    def __setitem__(self, indexer, value):
+        """Set transformation(s) at given index(es) in this object.
+
+        Parameters
+        ----------
+        indexer : int or slice or array_like
+            Specifies which transformation(s) to replace. A single indexer must
+            be specified, i.e. as if indexing a 1 dimensional array or list.
+
+        value : `RigidTransformation` instance
+            The transformation(s) to set.
+
+        Raises
+        ------
+        TypeError
+            If the transformation is a single transformation.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as Tf
+        >>> t = [[0, 0, 0], [1, 0, 0], [2, 0, 0]]  # 3 translations
+        >>> tf = Tf.from_translation(t)
+
+        Set a single transformation:
+
+        >>> tf[0] = Tf.from_translation([9, 9, 9])
+        >>> tf.translation
+        array([[9., 9., 9.],
+               [1., 0., 0.],
+               [2., 0., 0.]])
+        """
+        if self._single:
+            raise TypeError("Single transformation is not subscriptable.")
+
+        if not isinstance(value, self.__class__):
+            raise TypeError("value must be a RigidTransformation object")
+
+        arr = np.asarray(self._matrix)
+        arr[indexer] = value.as_matrix()
+        self._matrix = arr
+
+    @cython.embedsignature(True)
+    def __mul__(RigidTransformation self, RigidTransformation other):
+        """Compose this transformation with the other.
+
+        If `p` and `q` are two transformations, then the composition of 'q
+        followed by p' is equivalent to `p * q`. In terms of transformation
+        matrices, the composition can be expressed as
+        ``p.as_matrix() @ q.as_matrix()``.
+
+        In terms of translations and rotations, the composition when applied to
+        a vector ``v`` is equivalent to
+        ``p.translation + p.rotation.apply(q.translation)
+        + (p.rotation * q.rotation).apply(v)``.
+
+        Parameters
+        ----------
+        other : `RigidTransformation` instance
+            Object containing the transformations to be composed with this one.
+
+        Returns
+        -------
+        `RigidTransformation` instance
+            The composed transformation. This function supports composition of
+            multiple transformations at a time. The following cases are
+            possible:
+
+            - Either ``p`` or ``q`` contains a single transformation. In this
```

Comment:
It probably belongs to the main doc section.

Reply:
Moved